File: app/src/main/java/com/kotlin/loyal/impl/ObservableServer.java
/*
package com.kotlin.loyal.impl;

import com.kotlin.loyal.beans.ResultBean;
import com.kotlin.loyal.beans.WeatherBean;

import io.reactivex.Observable;
import okhttp3.MultipartBody;
import okhttp3.RequestBody;
import okhttp3.ResponseBody;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.Multipart;
import retrofit2.http.POST;
import retrofit2.http.Part;
import retrofit2.http.Streaming;
import retrofit2.http.Url;

import static com.kotlin.loyal.impl.Contact.Str.*;

public interface ObservableServer {

    @FormUrlEncoded
    @POST(INSTANCE.getAction() + "doLoginTest")
    Observable<String> doTestLogin(@Field("account") String account, @Field("password") String password);

    @FormUrlEncoded
    @POST(INSTANCE.getAction() + INSTANCE.getMethod_register())
    Observable<ResultBean> doRegister(@Field("json_register") String json);

    @FormUrlEncoded
    @POST(INSTANCE.getAction() + INSTANCE.getMethod_queryAccount())
    Observable<ResultBean> doQueryAccount(@Field("account") String account);

    @FormUrlEncoded
    @POST(INSTANCE.getAction() + INSTANCE.getMethod_login())
    Observable<ResultBean> doLogin(@Field("json_login") String json);

    @FormUrlEncoded
    @POST(INSTANCE.getAction() + INSTANCE.getMethod_destroyAccount())
    Observable<ResultBean> destroyAccount(@Field("json_destroy") String json);

    @FormUrlEncoded
    @Streaming
    @POST(INSTANCE.getAction() + INSTANCE.getMethod_showIcon())
    Observable<ResponseBody> doShowIcon(@Field("account") String account);

    @FormUrlEncoded
    @POST(INSTANCE.getAction() + INSTANCE.getMethod_update())
    Observable<ResultBean> doUpdateAccount(@Field("json_update") String json);

    @FormUrlEncoded
    @POST(INSTANCE.getAction() + INSTANCE.getMethod_account_locked())
    Observable<ResultBean> doAccountLocked(@Field("account") String account);

    @Multipart
    @POST(INSTANCE.getAction() + INSTANCE.getMethod_update_icon())
    Observable<String> doUpdateIcon(@Part("description") RequestBody description, @Part MultipartBody.Part iconFile);

    @FormUrlEncoded
    @POST(INSTANCE.getAction() + INSTANCE.getMethod_feedBack())
    Observable<ResultBean> doFeedBack(@Field("json_feed") String json);

    @FormUrlEncoded
    @POST(INSTANCE.getAction() + INSTANCE.getMethod_getSelfFeed())
    Observable<ResultBean> getSelfFeed(@Field("account") String account);

    @FormUrlEncoded
    @POST(INSTANCE.getAction() + INSTANCE.getMethod_deleteSelfFeed())
    Observable<ResultBean> deleteSelfFeed(@Field("json_delete") String json);

    @FormUrlEncoded
    @POST(INSTANCE.getAction() + INSTANCE.getMethod_ucrop_test())
    Observable<String> doUCropTest(@Field("account") String account, @Field("password") String password);

    @FormUrlEncoded
    @POST(INSTANCE.getAction() + INSTANCE.getMethod_scan())
    Observable<ResultBean> doScan(@Field("json_scan") String json, @Field("param") String param);

    @FormUrlEncoded
    @POST(INSTANCE.getAction() + INSTANCE.getMethod_apkVerCheck())
    Observable<ResultBean> doApkVer(@Field("apkVer") String apkVer);

    @Streaming
    @GET
    Observable<ResponseBody> doDownLoadApk(@Url String url);

    @GET
    Observable<WeatherBean> getWeather(@Url String url);
}*/

File: app/src/main/java/com/kotlin/loyal/libs/rxjava/ObservableServer.kt
package com.kotlin.loyal.libs.rxjava

import com.kotlin.loyal.beans.WeatherBean
import com.kotlin.loyal.impl.Contact.Str.action
import com.kotlin.loyal.impl.Contact.Str.method_account_locked
import com.kotlin.loyal.impl.Contact.Str.method_apkVerCheck
import com.kotlin.loyal.impl.Contact.Str.method_deleteSelfFeed
import com.kotlin.loyal.impl.Contact.Str.method_destroyAccount
import com.kotlin.loyal.impl.Contact.Str.method_feedBack
import com.kotlin.loyal.impl.Contact.Str.method_getSelfFeed
import com.kotlin.loyal.impl.Contact.Str.method_login
import com.kotlin.loyal.impl.Contact.Str.method_queryAccount
import com.kotlin.loyal.impl.Contact.Str.method_register
import com.kotlin.loyal.impl.Contact.Str.method_scan
import com.kotlin.loyal.impl.Contact.Str.method_showIcon
import com.kotlin.loyal.impl.Contact.Str.method_ucrop_test
import com.kotlin.loyal.impl.Contact.Str.method_update
import com.kotlin.loyal.impl.Contact.Str.method_update_icon
import io.reactivex.Observable
import okhttp3.MultipartBody
import okhttp3.RequestBody
import okhttp3.ResponseBody
import retrofit2.http.*

interface ObservableServer {

    @FormUrlEncoded
    @POST(action + method_register)
    fun doRegister(@Field("json_register") json: String): Observable<String>

    @FormUrlEncoded
    @POST(action + method_queryAccount)
    fun doQueryAccount(@Field("account") account: String): Observable<String>

    @FormUrlEncoded
    @POST(action + method_login)
    fun doLogin(@Field("json_login") json: String): Observable<String>

    @FormUrlEncoded
    @POST(action + method_destroyAccount)
    fun destroyAccount(@Field("json_destroy") json: String): Observable<String>

    @Streaming
    @GET
    fun doShowIcon(@Url url: String): Observable<ResponseBody>

    @Streaming
    @FormUrlEncoded
    @POST(action + method_showIcon)
    fun downIconByAccount(@Field("account") account: String): Observable<ResponseBody>

    @FormUrlEncoded
    @POST(action + method_update)
    fun doUpdateAccount(@Field("json_update") json: String): Observable<String>

    @FormUrlEncoded
    @POST(action + method_account_locked)
    fun doAccountLocked(@Field("account") account: String): Observable<String>

    @Multipart
    @POST(action + method_update_icon)
    fun doUpdateIcon(@Part("description") description: RequestBody, @Part iconFile: MultipartBody.Part): Observable<String>

    @FormUrlEncoded
    @POST(action + method_feedBack)
    fun doFeedBack(@Field("json_feed") json: String): Observable<String>

    @FormUrlEncoded
    @POST(action + method_getSelfFeed)
    fun getSelfFeed(@Field("account") account: String): Observable<String>

    @FormUrlEncoded
    @POST(action + method_deleteSelfFeed)
    fun deleteSelfFeed(@Field("json_delete") json: String): Observable<String>

    @FormUrlEncoded
    @POST(action + method_ucrop_test)
    fun doUCropTest(@Field("account") account: String, @Field("password") password: String): Observable<String>

    @FormUrlEncoded
    @POST(action + method_scan)
    fun doScan(@Field("json_scan") json: String, @Field("param") param: String): Observable<String>

    @FormUrlEncoded
    @POST(action + method_apkVerCheck)
    fun doApkVer(@Field("apkVer") apkVer: String): Observable<String>

    @Streaming
    @GET
    fun doDownLoadApk(@Url url: String): Observable<ResponseBody>

    @GET
    fun getWeather(@Url url: String): Observable<WeatherBean>
}
